src/bundle/Resources/public/js/scripts/embedded.item.actions.js

```diff
@@ -0,0 +1,318 @@
+                                      };
+
+                                return {
+                                    label: name,
+                                    ...languageEditAction,
+                                };
+                            }),
+                        },
+                    ],
+                },
+            };
+        }
+
+        const editAction = productCode
+            ? {
+                  href: Routing.generate('ibexa.product_catalog.product.edit', {
+                      productCode,
+                      languageCode: languages[0].languageCode,
+                  }),
+              }
+            : {
+                  onClick: () =>
+                      editContent({
+                          contentId,
+                          locationId,
+                          languageCode: languages[0].languageCode,
+                      }),
+              };
+
+        return {
+            label: Translator.trans(/*@Desc("Edit")*/ 'embedded_items.action.edit', {}, 'ibexa_content'),
+            ...editAction,
+        };
+    };
+    const generateMenuTreeItems = ({ contentId, locationId, productCode, languages }) => {
+        const goToItem = generateGoToActionItem({ contentId, locationId, productCode });
+        const editItem = generateEditActionItem({ contentId, locationId, productCode, languages });
+
+        return {
+            groups: [
+                {
+                    id: 'default',
+                    items: [goToItem, editItem],
+                },
+            ],
+        };
+    };
+    const getLanguagesData = async ({ contentId, initialFunc = () => {}, callbackFunc = () => {} }) => {
+        try {
+            initialFunc();
+
+            const url = window.Routing.generate('ibexa.permission.limitation.language.content_edit', { contentInfoId: contentId });
+            const request = new Request(url, {
+                method: 'GET',
+                headers: { 'X-CSRF-Token': token },
+                mode: 'same-origin',
+                credentials: 'same-origin',
+            });
+            const response = await fetch(request);
+            const data = await ibexa.helpers.request.getJsonFromResponse(response);
+
+            callbackFunc();
+
+            return data.filter((language) => language.hasAccess);
+        } catch (error) {
+            ibexa.helpers.notification.showErrorNotification(error);
+        }
+    };
+    const getMenuData = ({ container, event }) => {
+        const { contentId, locationId, productCode, languageCodes } = container ? container.dataset : event.detail;
+        const parsedLanguageCodes = typeof languageCodes === 'string' ? JSON.parse(languageCodes) : languageCodes;
+        const languages = parsedLanguageCodes
+            ? parsedLanguageCodes.map((languageCode) => ({
+                  languageCode,
+                  name: adminUiLanguages[languageCode].name,
+              }))
+            : [];
```

Comment:
```suggestion
        const { contentId, locationId, productCode, languageCodes = [] } = container ? container.dataset : event.detail;
        const parsedLanguageCodes = typeof languageCodes === 'string' ? JSON.parse(languageCodes) : languageCodes;
        const languages = parsedLanguageCodes.map((languageCode) => ({
              languageCode,
              name: adminUiLanguages[languageCode].name,
          }));
```
wdyt? It's up to you :)